dataikuapi/dss/apideployer.py

```diff
@@ -191,6 +200,18 @@ def add_apinode(self, url, api_key, graphite_prefix=None):
         }
         self.settings["apiNodes"].append(new_node)
 
+    def remove_apinode(self, node_url):
+        """
+        Removes a node from the list of nodes of this infra.
+        Only applicable to STATIC infrastructures
+
+        :param str node_url: URL of the node to remove
+        """
+        for node in self.settings["apiNodes"]:
+            if node["url"] == node_url:
```

Comment:
URL-empty proof by using .get("url", None), prevent some errors.

Reply:
It was my first move, but in `add_apinode` brackets are used instead of `get` as well. Actually we could even try/except the KeyError and have a message to explain that the infra is of the wrong type?

Reply:
Meh, no need to go that far, would introduce tight coupling. Using `get` is error free and will do what is requested (if the infra has the wrong type, you still removed every node with that url, even if it means none).

Reply:
Actually I read to fast and thought we were speaking of `self.settings["apiNodes"]`, do we want to use `get` here as well?

Reply:
apiNodes is defaulted to be empty list for a static infra settings, so no need to change that, the content of the list, at contrary, can be anything since it's a user input

Reply:
Only if the infra is a static one and not a K8S though right? But well the user is not supposed to use this if his infra is not staticanyway so this might be overkill

Reply:
That's in line with what's done in add, so indeed let's not overthink it